launch/launch/launcher.py

```diff
@@ -225,7 +254,18 @@ def _run(self):
                         except ProcessLookupError:
                             pass
 
-            yield from asyncio.wait(all_futures.keys())
+            kwargs = {}
+            if not isinstance(threading.current_thread(), threading._MainThread):
+                # wake up periodically if we are not in the main thread
```

Comment:
If the function is *not* invoked in the main thread then this condition should be satisfied and the timeout argument added, as far as I can see. I don't think there is any semantic ambiguity on what the "_MainThread" is, so I don't see how there can be a false negative?